### PR TITLE
build: add test_bitcoin.exe to win installer, don't install entire docs/ dir

### LINKED ISSUE #17171: windows: Files installed by installer *-setup.exe
The installer [`bitcoin-0.19.0rc1-win64-setup.exe`](https://bitcoincore.org/bin/bitcoin-core-0.19.0/test.rc1/bitcoin-0.19.0rc1-win64-setup.exe):

- lacks `test_bitcoin.exe` (is present in `zip`)

- bloats system with _full_ `doc` folder content:
Screenshot from 2019-10-17 11-03-30

### PR DESCRIPTION
**Pull request by fanquake**
Haven't tested other than checking that it Guix builds.

Fixes: #17171.

Guix build (x86_64):
```bash
6e2886c80eba9c829047c04586b142d5f8f1c53c31aa82834aff39ae5dbf1762  guix-build-d755ffc3277c/output/dist-archive/bitcoin-d755ffc3277c.tar.gz
cdf727c45c3283523726b4ec27f051de5931469874af736eac05d48016d6369b  guix-build-d755ffc3277c/output/x86_64-w64-mingw32/SHA256SUMS.part
546866b2f0c8067c168a936246c4cda25745c1b484322201230b885511f2abd7  guix-build-d755ffc3277c/output/x86_64-w64-mingw32/bitcoin-d755ffc3277c-win64-debug.zip
31dbb780dff003089d0e9a3a2598cde89453af4f1b18e392a186a6ec14718b48  guix-build-d755ffc3277c/output/x86_64-w64-mingw32/bitcoin-d755ffc3277c-win64-setup-unsigned.exe
39f1c55a2426390f014282d0a736ceb77e461199fde6ccefcef53ecf10dc4960  guix-build-d755ffc3277c/output/x86_64-w64-mingw32/bitcoin-d755ffc3277c-win64-unsigned.tar.gz
7e4f7dc3475598d187e77cc31842ad2ce876fb98dc42e999b32bdefbf0b79df1  guix-build-d755ffc3277c/output/x86_64-w64-mingw32/bitcoin-d755ffc3277c-win64.zip
```

### CODE REVIEW

**Comment by 1440000bytes** [comment]
Concept ACK

I could test if it was a change possible to test using windows binaries built using instructions in https://github.com/bitcoin/bitcoin/blob/master/doc/build-windows.md

No clue about guix build process

File: share/setup.nsi.in [hunk 1]
```diff
@@ -80,8 +80,7 @@ Section -Main SEC0000
     File @abs_top_builddir@/release/@BITCOIN_CLI_NAME@@EXEEXT@
     File @abs_top_builddir@/release/@BITCOIN_TX_NAME@@EXEEXT@
     File @abs_top_builddir@/release/@BITCOIN_WALLET_TOOL_NAME@@EXEEXT@
-    SetOutPath $INSTDIR\doc
-    File /r /x Makefile* @abs_top_srcdir@/doc\*.*
```
Comment (luke-jr) [inline share/setup.nsi.in:84]: Do we not want to install *any* docs?

**Review by jarolrod — APPROVED** [review]
ACK d755ffc3277c11b5c4a38c52e900b561d1636d0b

Tested on Windows 10 (x86) and Windows 11 (x86 and arm64). Was able to successfully run the tests in all scenarios, and run `bitcoin-qt`. Confirming that the docs dir is not placed in the installation directory.

File: share/setup.nsi.in (hunk 1 above)
Comment (jarolrod) [reply share/setup.nsi.in:84]: what would be docs that would be critical to include with the installation executable?

**Comment by fanquake** [comment]
> I could test if it was a change possible to test with windows binaries built using instructions in https://github.com/bitcoin/bitcoin/blob/master/doc/build-windows.md

It is. Run `make deploy` and that will build the installer.

File: share/setup.nsi.in (hunk 1 above)
Comment (fanquake) [reply share/setup.nsi.in:84]: > Do we not want to install any docs?

Windows is the odd one out in this regard. I think the README and copyright is fine. I'd assume most users never even realised we were dumping a bumping of random/non-windows documentation onto their drives. If you think we should be installing some, then you should suggest a list.

**Review by ghost — APPROVED** [review]
ACK https://github.com/bitcoin/bitcoin/pull/25809/commits/d755ffc3277c11b5c4a38c52e900b561d1636d0b

Tested on Win 10 and installer works as expected. Docs directory isn't added and test_bitcoin.exe is added in daemon directory.

```
PS C:\Program Files\Bitcoin> ls


    Directory: C:\Program Files\Bitcoin


Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
d-----        10-08-2022     16:40                daemon
-a----        10-08-2022     16:42       36432928 bitcoin-qt.exe
-a----        10-08-2022     16:42           1142 COPYING.txt
-a----        10-08-2022     16:42            846 readme.txt
-a----        10-08-2022     16:42         152493 uninstall.exe


PS C:\Program Files\Bitcoin> ls .\daemon\


    Directory: C:\Program Files\Bitcoin\daemon


Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a----        10-08-2022     16:42        2176512 bitcoin-cli.exe
-a----        10-08-2022     16:42        3810304 bitcoin-tx.exe
-a----        10-08-2022     16:42        8976896 bitcoin-wallet.exe
-a----        10-08-2022     16:42       14503424 bitcoind.exe
-a----        10-08-2022     16:42       24411648 test_bitcoin.exe
```

---

Although bitcoin core (windows) users will have to manually delete docs directory because installer just ensures its not added with new docs, it does not delete the docs already present.

File: share/setup.nsi.in (hunk 1 above)
Comment (luke-jr) [reply share/setup.nsi.in:84]: It's not a question of critical, just what's appropriate. I would think most of the docs would be. Exceptions are build docs, old release notes, and similar.

Comment (fanquake) [reply share/setup.nsi.in:84]: >  I would think most of the docs would be. Exceptions are build docs, old release notes, and similar.

Again, feel free to propose a list / change. I think your assessment of "most" is incorrect. Of all the `.md` files in the 23.x `docs/` folder, less than half would be relevent.

In any case don't think this something worth spending time to try and optimise. If removing the installation of docs is too contreversial, happy to just drop that commit from this PR, and we can close the second half of #17171 as wont fix.

**Comment by hebasto** [comment]
Guix build on `x86_64`:
```
6e2886c80eba9c829047c04586b142d5f8f1c53c31aa82834aff39ae5dbf1762  guix-build-d755ffc3277c/output/dist-archive/bitcoin-d755ffc3277c.tar.gz
cdf727c45c3283523726b4ec27f051de5931469874af736eac05d48016d6369b  guix-build-d755ffc3277c/output/x86_64-w64-mingw32/SHA256SUMS.part
546866b2f0c8067c168a936246c4cda25745c1b484322201230b885511f2abd7  guix-build-d755ffc3277c/output/x86_64-w64-mingw32/bitcoin-d755ffc3277c-win64-debug.zip
31dbb780dff003089d0e9a3a2598cde89453af4f1b18e392a186a6ec14718b48  guix-build-d755ffc3277c/output/x86_64-w64-mingw32/bitcoin-d755ffc3277c-win64-setup-unsigned.exe
39f1c55a2426390f014282d0a736ceb77e461199fde6ccefcef53ecf10dc4960  guix-build-d755ffc3277c/output/x86_64-w64-mingw32/bitcoin-d755ffc3277c-win64-unsigned.tar.gz
7e4f7dc3475598d187e77cc31842ad2ce876fb98dc42e999b32bdefbf0b79df1  guix-build-d755ffc3277c/output/x86_64-w64-mingw32/bitcoin-d755ffc3277c-win64.zip
```

**Review by hebasto — APPROVED** [review]
ACK d755ffc3277c11b5c4a38c52e900b561d1636d0b, tested on Windows 11 Pro.

Should the `24.0` milestone be assigned here?

**Comment by fanquake** [comment]
> Should the 24.0 milestone be assigned here?

I don't think this needed a milestone, but is going to be merged now in any case. I'll combine some other minor share additions into #25829.

**Pull request merged**